kroxylicious/src/main/java/io/kroxylicious/proxy/internal/DefaultFilterContext.java

```diff
@@ -36,17 +36,20 @@ class DefaultFilterContext implements KrpcFilterContext {
     private final ChannelPromise promise;
     private final KrpcFilter filter;
     private final long timeoutMs;
+    private final String sniHostname;
 
     DefaultFilterContext(KrpcFilter filter,
                          ChannelHandlerContext channelContext,
                          DecodedFrame<?, ?> decodedFrame,
                          ChannelPromise promise,
-                         long timeoutMs) {
+                         long timeoutMs,
+                         String sniHostname) {
```

Comment:
We should think if there are other connection-related things that a protocol filter might reasonably need access to. If there are many we might want to use some object to pass them through the layers, so we avoid needing to add fields and constructor parameters through the layers every time we want to add something from the connection to the `FilterContext`.

Reply:
I was wondering about an `UpstreamContext` object?  As well as SNI, it might expose things like the client IP, the TLS protocol etc.

Reply:
@tombentley WDYT to this idea?